File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

public class Gigasecond {
    public Gigasecond(LocalDate moment) {
      moment_ = moment.atTime(0, 0);
    }

    public Gigasecond(LocalDateTime moment) {
      moment_ = moment;
    }

    public LocalDateTime getDateTime() {
      return moment_.plusSeconds(1000000000);  // one Gigasecond added
    }
  
  private LocalDateTime moment_;
}

File: kotlin/gigasecond/src/main/kotlin/Gigasecond.kt
import java.time.LocalDateTime
import java.time.LocalDate
import java.time.LocalTime

class Gigasecond(input_date: LocalDate) {
  
  val one_gigasecond: LocalTime = LocalTime.ofSecondOfDay(1000000000)
  // TODO: Implement proper constructor
  val date: LocalDateTime = input_date.atTime(one_gigasecond) 
}
